### DISCUSSION TITLE
Are storage and build cache always in the same directory?

### DISCUSSION THREAD

**Discussion by nschan** [hpc/charliecloud#1922] [General]
Hello,

I understand that if `CH_IMAGE_STORAGE` is set and I run e.g. `ch-image pull someimage` the cache will be in `$CH_IMAGE_STORAGE`. 

If I set `$CH_IMAGE_STORAGE` and ran `ch-image pull -s /some/dir someimage` (`/some/dir` not being the same directory as `$CH_IMAGE_STORAGE`), since `-s` takes priority over `$CH_IMAGE_STORAGE`, I assume the build cache would be in `/some/dir`? 

Is there a way to e.g. pull an image to `/some/dir`, while making use of the cache in `$CH_IMAGE_STORAGE`?

Thanks a lot
Niklas

**Comment by reidpr** [comment]
Hello Niklas,

Not at this time, no. But does #1353 match your request?

HTH,
Reid

**Reply by nschan** [reply]
That would actually be quite helpful I think. 
The issue that I need to work around is pulling images that can be run with `-w` (preferable without an intermediate convert) while making use of a central cache.

**Reply by reidpr** [reply]
Couple suggestions would be:

1. `ch-convert`, as you note. At present this just does a normal copy, though in principle it could do a reflink (metadata-only) copy, which would be quite a bit faster given appropriate filesystem support.
2. `-W`/`--write-fake`, which I think works with images in the storage directory?

**Reply by nschan** [reply]
`--write-fake` would be the preferred way I guess, but the cluster I am using is on an ancient kernel (4.12 iirc), so that's not an option. 

With the way `convert` currently works, convert and `pull` take roughly the same time for me, so I would kind of prefer to just pull. 

I think with the cache being in the same place as is used for storage with `-s`, the current implementation is actually ok for me. I can simply maintain a storage for images that are run with `-w`, separate from `$CH_IMAGE_STORAGE`.
